src/main/kotlin/org/rust/lang/core/types/ty/TyTypeParameter.kt

```diff
@@ -34,7 +36,27 @@ class TyTypeParameter private constructor(
         bounds.flatMap { it.flattenHierarchy }
 
     override fun canUnifyWith(other: Ty, project: Project, mapping: TypeMapping?): Boolean {
```

Comment:
You can rename this method (mentally, of course) to `tryUnifyWith` and `mapping` to something like `unificationTable`. In fact, if we checked a trait bounds, there would be a very similar code insted of `return true`